src/test/scala/chiselTests/util/SizeOfSpec.scala

```diff
@@ -0,0 +1,33 @@
+package chiselTests.util
+
+import circt.stage.CIRCTStage
+
+import java.io.File
+import java.io.{PrintWriter, Writer}
+
+import chisel3._
+import chisel3.stage.ChiselStage
+import chisel3.testers.BasicTester
+import chisel3.util.SizeOf
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.Source
+
+class SizeOfTop extends Module {
+  val io = IO(new Bundle{
+    val w = Input(UInt(65.W))
+    val outw = UInt(32.W)
+  })
+  io.outw  := SizeOf(io.w)
+}
+
+/** A test for intrinsics */
+class SizeOfSpec extends AnyFlatSpec with Matchers {
+  it should "Should work for types" in {
+    val fir = circt.stage.ChiselStage.emitFIRRTLDialect(new SizeOfTop)
```

Comment:
What if the way this is generated changes but the behavior stays compliant? What if the syntax of the CIRCT firrtl IR changes? What about non-CIRCT firrtl compilers?
If you generate a `BasicTester` then this has the advantage that the test will work with any firrtl compliant compiler.

Reply:
Chisel generates firrtl, so what matters is that the firrtl + annotations are what they should be.  There is no reason to repeat tests which already exist in firrtl compilers for the actual implementation of the intrinsic.

The test is updated to just check the firrtl is the expected form.

When the spec has explicit intrinisics,  we'll check those are generated correctly.

Reply:
> Chisel generates firrtl, so what matters is that the firrtl + annotations are what they should be.

This test does not check standard firrtl, but some internal CIRCT dialect instead.
Also, you are not checking the firrtl generated by Chisel (which would not be as brittle), but the firrtl generated after running some CIRCT compiler passes, which means that if CIRCT ever decides to represent the result differently, i.e. a different constant encoding or somehow adding an intermediate signal, then this test would break.

Reply:
Not sure how many Chisel tests you have so far ported to CIRCT, however, I assume that all the execution tests will be easy to port - you just need to change the way the Verilog is generated - while any test that matches on the exact Verilog strings being generated will require manual effort to port. Thus executable tests are much more robust wrt. future compiler changes.

Reply:
We have lots of tests that check emitted .fir or .v. There is already a PR open by @chick starting to port them over.

> The test is updated to just check the firrtl is the expected form.

@ekiwi are you saying this is still not the case, since you last reviewed it?

Reply:
> are you saying this is still not the case, since you last reviewed it?

Ah sorry, somehow github was showing me the old code. I think the new version is an improvement. I am still a big fan of executable tests where possible, but the new version is fine since it is a lot more robust wrt. firrtl compiler improvements.

Reply:
>  Thus executable tests are much more robust wrt. future compiler changes.

Not here, since not all compilers or compiler versions are expected to support an intrinsic.  Executable tests are testing both the generation which is compiler and version independent and the implementation which is compiler and version dependent.  Just testing that the correct firrtl is generated is more specific and more robust.